Refactor from More Features to Embeds, Events, Markdown

Elitesparkle opened the pull request:
Split the [More Features](https://guide.pycord.dev/getting-started/more-features) page into 3 separate pages: Embeds, Events, Markdown. Furthermore, add Related Topics to connect the Embeds page and the Markdown page to each other.

Elitesparkle committed:
Refactor from More Features to Embeds, Events, Markdown

Elitesparkle committed:
Fix end of files

BobDotCom committed:
New category + md examples

BobDotCom commented:
I added a couple markdown examples, and moved the element style pages into their own category.

BobDotCom commented on docs/getting-started/styling-messages/markdown.mdx:
````diff
@@ -0,0 +1,248 @@
+  <DiscordMessage profile="robocord">
+    <DiscordMarkdown>
+```
+message = "Hello world!"
+print(message)
````
Note: this doesn't format correctly at the moment. Fix before merging. May be resolved by #239.

BobDotCom committed:
Merge branch 'main' into more-features

Elitesparkle committed:
Fix syntax highlight, sections, examples, and some sentences

Add the "text" flag to code blocks that aren't supposed to show with syntax highlight as they are about an example without it.

Split the Code Block and Quote Block sections into paragraphs.

Add an example for single-line code block with syntax highlight.

Improve some sentences here and there.

Elitesparkle committed:
Merge branch 'main' into more-features

Elitesparkle committed:
Merge branch 'main' into more-features

Elitesparkle commented:
There are 3 warnings about code blocks. I'm not sure if they need to be changed and how.

Elitesparkle committed:
Fix internal link

Elitesparkle committed:
Fix examples

Elitesparkle committed:
Refactor indentation

Elitesparkle committed:
Merge branch 'main' into more-features

Elitesparkle committed:
Fix multi-line quote example

The syntax isn't exactly the one explained immediately above but the final result looks exactly the same and we hopefully no longer get any warning there.

Elitesparkle committed:
Merge branch 'more-features' of https://github.com/Elitesparkle/guide into more-features

Elitesparkle committed:
Revert "Fix examples"

Elitesparkle committed:
Fix examples

Elitesparkle committed:
Fix titles and multi-line quote example

Elitesparkle committed:
Remove non-working language names from code block examples

Elitesparkle commented on docs/getting-started/styling-messages/markdown.mdx:
````diff
@@ -0,0 +1,284 @@
+<DiscordComponent>
+  <DiscordMessage profile="robocord">
+    <DiscordMarkdown>
+```
````
If we want to show this example properly, this code block should be rendered with syntax highlight. Since it was showing "python" as text, I reverted the change for now. The next 2 examples have the same problem.

BobDotCom replied:
```suggestion
```python
```

Elitesparkle commented on docs/getting-started/styling-messages/markdown.mdx:
```diff
@@ -0,0 +1,284 @@
+<DiscordComponent>
+  <DiscordMessage profile="robocord">
+    <DiscordMarkdown>
+> This is a
```
If we write `>>>`, we get an error. If we write `>` for each line, the first line shows as quote block correctly but the second `>` shows as text instead of being rendered as quote block.

BobDotCom commented on docs/getting-started/styling-messages/markdown.mdx:
````diff
@@ -0,0 +1,284 @@
+<DiscordComponent>
+  <DiscordMessage profile="robocord">
+    <DiscordMarkdown>
+```
````
```suggestion
```python
```

BobDotCom commented on docs/getting-started/styling-messages/markdown.mdx:
```diff
@@ -0,0 +1,284 @@
+<DiscordComponent>
+  <DiscordMessage profile="robocord">
+    <DiscordMarkdown>
+> This is a
+> multi-line quote.
```
```suggestion
>>> This is a
multi-line quote.
```

BobDotCom reviewed (commented):
We should always write the correct syntax, and change the discordcomponent if it fails to parse correctly

BobDotCom committed:
Update docs/getting-started/styling-messages/markdown.mdx

BobDotCom committed:
Update docs/getting-started/styling-messages/markdown.mdx

BobDotCom committed:
Update docs/getting-started/styling-messages/markdown.mdx

Louis790 commented on docs/getting-started/styling-messages/embeds.mdx:
```diff
@@ -201,9 +146,9 @@ embed.add_field(title="Inline Field 2", value="Inline Field 2", inline=True)
+This small section shows off embed fields. You can add up to 25 fields to embeds with the [`add_field` method](https://docs.pycord.dev/en/master/api.html#discord.Embed.add_field)
 of the [`discord.Embed`](https://docs.pycord.dev/en/master/api.html#embed) class. These consist of three 
-keyword arguments: `title`, `value`, and `inline`. `title` and `value` are both required arguments, which inline defaults to `False` if it's not defined. The `inline` argument specifies whether or not space will be divided among the inline fields. There could be a mix of fields where inline has different values too.
+keyword arguments: `title`, `value`, and `inline`. `title` and `value` are both required arguments, whereas `inline` is optional and defaults to `False` when not defined. The `inline` argument specifies whether or not space will be divided among the inline fields. There could be a mix of fields where inline has different values too.
```
According to [the docs](https://docs.pycord.dev/en/stable/api/data_classes.html#discord.Embed.add_field), inline defaults to True.
```suggestion
keyword arguments: `title`, `value`, and `inline`. `title` and `value` are both required arguments, whereas `inline` is optional and defaults to `True` when not defined. The `inline` argument specifies whether or not space will be divided among the inline fields. There could be a mix of fields where inline has different values too.
```

Elitesparkle committed:
Update default value for inline field in embeds

Co-authored-by: Louis <80314632+Louis790@users.noreply.github.com>
Signed-off-by: Elitesparkle <Elitesparkle@users.noreply.github.com>
